test/e2e_node/critical_pod_test.go

```diff
@@ -142,9 +145,9 @@ func getTestPod(critical bool, name string, resources v1.ResourceRequirements) *
 		pod.ObjectMeta.Annotations = map[string]string{
 			kubelettypes.CriticalPodAnnotationKey: "",
 		}
-		Expect(kubelettypes.IsCriticalPod(pod)).To(BeTrue(), "pod should be a critical pod")
+		Expect(kubelettypes.IsCritical(pod.Namespace, pod.Annotations)).To(BeTrue(), "pod should be a critical pod")
```

Comment:
why change these two?  `IsCritical()` is called by `IsCriticalPod()` in the case that the `ExperimentalCriticalPodAnnotation` feature gate is enabled.

Reply:
I see.  But... aren't you making the code tolerant of this situation with the change on line 48?

Reply:
The code change on line 48 is for a different problem.  If you run the test without any additional feature-gates enabled, the test will panic, as the feature gates map is empty.
Using dynamic kubelet config to enable feature gates on the kubelet does not enable them in the test framework, so the change on line 48 won't affect the `IsCriticalPod` call during the test. Maybe the better change is to remove this check all-together, as it is something that should be done in a unit-test.

Reply:
or maybe add the check for a nil feature gate map in `IsCriticalPod()`?

Reply:
The change on line 48 prevents a panic during the test if not supplying _any additional_ feature gates.  This is required regardless of the `IsCriticalPod` check on line 145, as tests shouldn't panic regardless of the feature gates initially set.

The change on line 145 (here) allows the test to succeed without specifying the `ExperimentalCriticalPodAnnotation` before running the test.  This removes a confusing test failure if the feature gate isn't specified.  Alternatively, we could either remove the check here entirely, or explicitly check that the feature gate is enabled to make the failure less confusing to track down.  The latter just makes the test more annoying for people to run.